website/static/playground.js

```diff
@@ -29,7 +29,12 @@ var IDEMPOTENT_MESSAGE = "✓ Second format is unchanged.";
 var state = (function loadState(hash) {
   var parsed;
   try {
-    parsed = JSON.parse(hash);
+    // providing backwards support for old json encoded URIComponent
+    if (hash.indexOf("%7B%22") !== -1) {
```

Comment:
%7B%22 is "{"

Reply:
Our playground url seems always prefixed with `{`, I think we can just check the first 6 char?

Reply:
@ikatyang thats true we could do that.  We could convert the check to

```javascript
hash.startsWith("%7B%22")
```

My concern is that would exclude older IE browsers (IE 12<). 

Do you have any other suggestion ?

Reply:
`hash.slice(0, 6) === "%7B%22"`?

Reply:
@ikatyang  is that be more perfomant than doing indexOf and regex ? (I haven't tried it out yet)

Reply:
I doubt there will be any performance difference no matter what approach you take.

Reply:
just for curiosity created this https://jsperf.com/indexof-vs-test-vs-slice/1 looks like indexOf is a bit faster when that is existent on the url

Reply:
Yeah, the browser optimization is crazy... I can't think of why it happened. 😵


![image](https://user-images.githubusercontent.com/8341033/31756331-10f83f0c-b4d6-11e7-9ba8-009e56b47fd1.png)

---

https://jsperf.com/string-startswith/3

![image](https://user-images.githubusercontent.com/8341033/31756281-cecf1e2a-b4d5-11e7-8587-57f4b7dc8063.png)

Reply:
So based on that will leave that as it is then with the current `indexOf` check.